Corona-Warn-App/src/main/java/de/rki/coronawarnapp/qrcode/handler/DccQrCodeHandler.kt

```diff
@@ -17,21 +18,23 @@ class DccQrCodeHandler @Inject constructor(
     private val vaccinationRepository: VaccinationRepository,
     private val testCertificateRepository: TestCertificateRepository,
     private val recoveryCertificateRepository: RecoveryCertificateRepository,
-    private val dscSignatureValidator: DscSignatureValidator
+    private val dscSignatureValidator: DscSignatureValidator,
+    private val boosterCheckScheduler: BoosterCheckScheduler,
 ) {
 
     /**
      * Saves [DccQrCode] in the respective repository after validating the signature
      * throws [InvalidHealthCertificateException]
      */
-    suspend fun handleQrCode(dccQrCode: DccQrCode):
-        CertificateContainerId {
-            dscSignatureValidator.validateSignature(dccData = dccQrCode.data)
-            return when (dccQrCode) {
-                is RecoveryCertificateQRCode -> recoveryCertificateRepository.registerCertificate(dccQrCode).containerId
-                is VaccinationCertificateQRCode -> vaccinationRepository.registerCertificate(dccQrCode).containerId
-                is TestCertificateQRCode -> testCertificateRepository.registerCertificate(dccQrCode).containerId
-                else -> throw UnsupportedQrCodeException()
-            }
+    suspend fun handleQrCode(dccQrCode: DccQrCode): CertificateContainerId {
+        dscSignatureValidator.validateSignature(dccData = dccQrCode.data)
+        return when (dccQrCode) {
+            is RecoveryCertificateQRCode -> recoveryCertificateRepository.registerCertificate(dccQrCode).containerId
+            is VaccinationCertificateQRCode -> vaccinationRepository.registerCertificate(dccQrCode).containerId
+            is TestCertificateQRCode -> testCertificateRepository.registerCertificate(dccQrCode).containerId
+            else -> throw UnsupportedQrCodeException()
+        }.also {
+            boosterCheckScheduler.scheduleNow()
         }
```

Comment:
Is it possible to move booster check into registerCertificate() so the workaround with the 3000 ms delay is not needed?

Reply:
Unfortunately it is not possible for the following reason:
- it creates a cyclic DI dependencies between BoosterNotificationService  <-> VaccinationRepository

Reply:
🤷‍♀️